### PR TITLE
MM-18024 MM-18025 Fix race condition when push notification opens the…

### PR DESCRIPTION
**Pull request by enahum**
… app and credentials are not set

Needs manual cherry-pick for 1.22

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18024
https://mattermost.atlassian.net/browse/MM-18025

### CODE REVIEW

File: app/components/post_list/post_list.js [hunk 1]
```diff
@@ -315,6 +318,20 @@ export default class PostList extends PureComponent {
         }
     };
 
+    scrollToInitialIndexIfNeeded = (index, count = 0) => {
+        if (!this.hasDoneInitialScroll && this.flatListRef?.current) {
+            this.hasDoneInitialScroll = true;
+
+            if (index > 0 && index <= this.getItemCount()) {
+                this.scrollToIndex(index);
+            } else if (count < 3) {
```
Comment (sudheerDev) [inline app/components/post_list/post_list.js]: what is the significance if count 3 here?

Comment (enahum) [reply app/components/post_list/post_list.js]: I wanted to retry 3 times before giving up

**Comment by enahum** [comment]
@hmhealey that is absolutely correct

**Pull request merged**